pkg/distributor/http.go

```diff
@@ -74,3 +76,28 @@ func (d *Distributor) PushHandler(w http.ResponseWriter, r *http.Request) {
 		serverutil.JSONError(w, http.StatusInternalServerError, err.Error())
 	}
 }
+
+// ServeHTTP implements the distributor ring status page.
+//
+// If the rate limiting strategy is local instead of global, no ring is used by
+// the distributor and as such, no ring status is returned from this function.
+func (d *Distributor) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if d.rateLimitStrat == validation.GlobalIngestionRateStrategy {
+		d.distributorsRing.ServeHTTP(w, r)
+		return
+	}
+
+	var noRingPage = `
+			<!DOCTYPE html>
+			<html>
+				<head>
+					<meta charset="UTF-8">
+					<title>Distributor Ring Status</title>
+				</head>
+				<body>
+					<h1>Distributor Ring Status</h1>
+					<p>Not running with Global Rating Limit - ring not being used by the Distributor.</p>
+				</body>
+			</html>`
+	cortex_util.WriteHTMLResponse(w, noRingPage)
```

Comment:
@DylanGuedes Can you please copy the `WriteHTMLResponse` to our loki's `util` package and use that instead? 
Rationale is to avoid introducing anymore dependency from Cortex. Also we are trying to remove existing dependencies incrementally.

Particularly this [PR ](https://github.com/grafana/loki/pull/4917)tries to remove `cortex/util` package dependency completely.

Reply:
Oops, good catch! I just pushed a new commit fixing that, lmk what you think.